testing/simulator/src/basic_sim.rs

```diff
@@ -246,41 +233,55 @@ pub fn run_eth1_sim(matches: &ArgMatches) -> Result<(), String> {
             // Check that all nodes have transitioned to the required fork.
             checks::verify_fork_version(
                 network.clone(),
-                if post_merge_sim {
-                    Epoch::new(BELLATRIX_FORK_EPOCH)
-                } else {
-                    Epoch::new(ALTAIR_FORK_EPOCH)
-                },
+                Epoch::new(latest_fork_start_epoch),
                 slot_duration,
-                if post_merge_sim {
-                    bellatrix_fork_version
-                } else {
-                    altair_fork_version
-                }
+                latest_fork_version,
             ),
             // Check that all sync aggregates are full.
             checks::verify_full_sync_aggregates_up_to(
                 network.clone(),
                 // Start checking for sync_aggregates at `FORK_EPOCH + 1` to account for
                 // inefficiencies in finding subnet peers at the `fork_slot`.
-                Epoch::new(ALTAIR_FORK_EPOCH + 1).start_slot(MinimalEthSpec::slots_per_epoch()),
-                Epoch::new(END_EPOCH).start_slot(MinimalEthSpec::slots_per_epoch()),
+                Epoch::new(ALTAIR_FORK_EPOCH + 1).start_slot(slots_per_epoch),
+                Epoch::new(END_EPOCH).start_slot(slots_per_epoch),
                 slot_duration,
             ),
             // Check that the transition block is finalized.
             checks::verify_transition_block_finalized(
                 network.clone(),
-                Epoch::new(TERMINAL_BLOCK / MinimalEthSpec::slots_per_epoch()),
+                Epoch::new(TERMINAL_BLOCK / slots_per_epoch),
                 slot_duration,
-                post_merge_sim
+                true,
             ),
             checks::verify_light_client_updates(
                 network.clone(),
                 // Sync aggregate available from slot 1 after Altair fork transition.
-                Epoch::new(ALTAIR_FORK_EPOCH).start_slot(MinimalEthSpec::slots_per_epoch()) + 1,
-                Epoch::new(END_EPOCH).start_slot(MinimalEthSpec::slots_per_epoch()),
+                Epoch::new(ALTAIR_FORK_EPOCH).start_slot(slots_per_epoch) + 1,
+                Epoch::new(END_EPOCH).start_slot(slots_per_epoch),
                 slot_duration
-            )
+            ),
+            checks::verify_full_blob_production_up_to(
+                network.clone(),
+                // Blobs should be available from the first slot after the Deneb fork.
+                Epoch::new(DENEB_FORK_EPOCH + 1).start_slot(slots_per_epoch) + 1,
```

Comment:
Don't think the offset should exist here
```suggestion
                Epoch::new(DENEB_FORK_EPOCH).start_slot(slots_per_epoch),
```